examples/flutter_gallery/lib/demo/customized/home.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'initial_welcome.dart';
+import 'run_stats_content.dart';
+import 'snapping_scroll_physics.dart';
+
+const double _kStatsSectionPeekAmount = 70.0;
+
+class CustomizedDesign extends StatefulWidget {
+
+  static const String routeName = '/customized';
+
+  @override
+  _CustomizedDesignState createState() => _CustomizedDesignState();
+}
+
+class _CustomizedDesignState extends State<CustomizedDesign> with TickerProviderStateMixin {
+
+  ThemeData _themeData;
+  bool _animatedStats = false;
+  Widget _mainContentWidget;
+  final GlobalKey<RunStatsContentState> _statsContentKey = GlobalKey<RunStatsContentState>();
+  Widget _statsContentWidget;
+  final GlobalKey<InitialWelcomeState> _initialWelcomeKey = GlobalKey<InitialWelcomeState>();
+  Widget _initialContentWidget;
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _configureThemes();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _initialContentWidget ??= InitialWelcome(key: _initialWelcomeKey);
```

Comment:
The generally Flutter-y thing to do is to create widgets during build. Why create some of them here? The construction of an immutable is cheap.